flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -333,7 +333,7 @@ public void updateInputChannel(InputChannelDeploymentDescriptor icdd) throws IOE
 
 			InputChannel current = inputChannels.get(partitionId);
 
-			if (current.getClass() == UnknownInputChannel.class) {
+			if (current instanceof UnknownInputChannel) {
```

Comment:
Just to be on the safe side, you should also change this check in `#setInputChannel()` above. This way, we handle all sub-classes of `UnknownInputChannel` the same way as `UnknownInputChannel` itself

Reply:
agree